enhancements/cluster-validation-tool.md

```diff
@@ -0,0 +1,536 @@
+able to run the tool at various points in the deployment lifecycle including against:
+* my live production clusters
+* source (eg generic/templated) CRs in my planning labs
+* reference CRs provided by partner
+
+As a support engineer helping to triage/debug issues with a customer or partner I want to be able to
+compare the configuration of a cluster as captured in a support archive against a validated
+reference configuration.
+
+As a provider of reference configurations I want to be able to provide a new or updated validated
+reference configuration without the need to rebuild the validation tool.
+
+### Goals
+
+The following goals apply to the validation (diff) tool:
+1. Data driven. New and updated reference configurations do not require a new release of the tool.
+1. Can be used with new and updated reference configurations
+1. Can consume input configuration CRs from live cluster
+1. Can consume input configuration CRs from a support archive
+1. Can consume input from local filesystem (files, directories, etc)
+1. Will suppress diffs (not report as a drift) for runtime variable fields (status, managed
+   metadata, etc)
+1. Will suppress diffs (not report as a drift) for know user variable content as described by the
+   reference configuration
+1. Will show diffs (report as drift) for content in input configuration which does not match
+   reference
+1. Will show diffs (report as drift) for reference configuration content missing from input
+   configuration
+1. Will show diffs (informational) for content in input configuration which is not contained in
+   reference
+1. Allows comparison against one-of-several in reference configuration (ie an input configuration CR
+   compared against one of several optional implementations in the reference configuration)
+
+### Non-Goals
+
+1. Validation which goes beyond what is available in the configuration CRs – deeper
+   inspection/analysis is the domain of other tools such as Insights – see
+   [CNF-9064](https://issues.redhat.com/browse/CNF-9064).
+1. Validation of configuration CRs against CRD (ie validation of their correctness or ability to be
+   successfully applied to a cluster)
+
+## Proposal
+
+### Terminology
+
+* **drift** -- A significant delta/difference which needs to be brough into compliance or undergo further review/assessment
+* **deviation** -- A delta/difference which has been deemed acceptable through some approval process
+
+### Validation Tool Implementation
+
+The validation tool will operate similarly to a standard Linux `diff` tool which operates across a
+set of inputs (eg directory trees). The left hand side of the diff will be the selected reference
+configuration (see below for structure/contents of the reference) and the right hand side will be a
+collection of the user’s configuration CRs. The logical flow of the tool will be:
+1. User invokes the tool with the two inputs: `validationTool ./referenceConfig/ ./userConfig/`
+1. For each CR in `./userConfig`
```

Comment:
In case the input config source is a live cluster, is the tool really supposed to go through all CRs on the cluster or there would be some boundaries? Would it perhaps make sense to limit the search to objects with apiVersion and kind mentioned (included) in the reference configuration?